### PR TITLE
Display errors only for changed code 

### PR DESCRIPTION
**Pull request by faishal**
Related to https://github.com/humanmade/product-dev/issues/624

- Consider only `add` items from change
- Add comments in the review

### CODE REVIEW

**Comment by rmccue** [comment]
Can we ensure these changes are behind env var flags too? The behaviour for hmlinter shouldn't change, only for altis-review, so we'll need to ensure they're variable in that regard.

**Comment by faishal** [comment]
- When `CHECK_ANNOTATION_ONLY_RELATED` is not set : [run result](https://github.com/humanmade/altis-code-review/pull/5/checks?check_run_id=1841996120)

image

- When `CHECK_ANNOTATION_ONLY_RELATED` is set to `1` : [run result](https://github.com/humanmade/altis-code-review/pull/5/checks?check_run_id=1841998474) 

image

File: src/diff.js [hunk 1]
```diff
@@ -31,7 +31,9 @@ module.exports.getDiffMapping = async ( pushConfig, number, github ) => {
 			}
 			chunk.changes.forEach( change => {
 				position++;
-				mapping[ file.to ][ change.ln || change.ln2 ] = position;
+				if( change.type === "add" ) {
```
Comment (rmccue) [inline src/diff.js:34]:
```suggestion
				if ( change.type === "add" ) {
```

What about for removals? 🤔

File: src/format.js [hunk 2]
```diff
@@ -220,15 +228,27 @@ const formatReviewChange = ( lintState, mapping, comparison ) => {
...
+const formatAnnotations = ( state, baseUrl, mapping ) => {
 	const combined = combineLinters( state.results );
-
+	state.totals.skipped = 0;
+	const files = resultsByFile( combined );
 	const annotations = [];
 	Object.keys( combined ).forEach( file => {
 		const comments = combined[ file ];
 		comments.forEach( comment => {
+			if( process.env.CHECK_ANNOTATION_ONLY_RELATED ){
```
Comment (rmccue) [inline src/format.js:242]:
```suggestion
			if ( process.env.CHECK_ANNOTATION_ONLY_RELATED ){
```

File: src/hooks.js [hunk 3]
```diff
@@ -297,25 +300,18 @@ const onOpenPull = async context => {
...
 	github.pullRequests.createReview( {
 		owner,
 		repo,
 		number:    payload.number,
-		commit_id: commit,
+		// commit_id: commit,
 		body:      body,
-		event,
+		 comments,
+		 event
```
Comment (rmccue) [inline src/hooks.js:314]: Invalid indentation here.

```suggestion
		comments,
		event
```

File: src/hooks.js [hunk 4]
```diff
@@ -297,25 +300,18 @@ const onOpenPull = async context => {
...
+	const { body, event, comments } = formatReview( lintState, diffMapping );
 
 	// path, position, body
 
 	github.pullRequests.createReview( {
 		owner,
 		repo,
 		number:    payload.number,
-		commit_id: commit,
+		// commit_id: commit,
```
Comment (rmccue) [inline src/hooks.js:311]: Why is this removed?

**Review by rmccue — CHANGES_REQUESTED** [review]
It looks like this is still using the comment mode rather than the annotation mode?

File: src/diff.js (hunk 1 above)
Comment (faishal) [reply src/diff.js:34]: we don't have to review the code that has been removed in terms of security/ performance issues.

Comment (rmccue) [reply src/diff.js:34]: Hmm, true; I thought it might screw up the offsetting or not work for changes, but seems fine.

**Review by rmccue — APPROVED** [review]

**Pull request merged**